test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1

```diff
@@ -126,49 +126,104 @@ Describe "Import-Module for Binary Modules in GAC" -Tags 'CI' {
 
 Describe "Import-Module for Binary Modules" -Tags 'CI' {
 
-    It "PS should try to load the assembly from file path first" {
- $src = @"
-using System.Management.Automation;           // Windows PowerShell namespace.
-
-namespace ModuleCmdlets
-{
-  [Cmdlet(VerbsDiagnostic.Test,"BinaryModuleCmdlet1")]
-  public class TestBinaryModuleCmdlet1Command : Cmdlet
-  {
-    protected override void BeginProcessing()
-    {
-      WriteObject("BinaryModuleCmdlet1 exported by the ModuleCmdlets module.");
-    }
-  }
-}
+    BeforeAll {
+        $src = @"
+            using System.Management.Automation;           // Windows PowerShell namespace.
+            namespace ModuleCmdlets
+            {
+                [Cmdlet(VerbsDiagnostic.Test,"BinaryModuleCmdlet1")]
+                public class TestBinaryModuleCmdlet1Command : Cmdlet
+                {
+                    protected override void BeginProcessing()
+                    {
+                    WriteObject("BinaryModuleCmdlet1 exported by the ModuleCmdlets module.");
+                    }
+                }
+            }
 "@
+        Add-Type -TypeDefinition $src -OutputAssembly $TESTDRIVE\System.dll
+        Add-Type -TypeDefinition $src -OutputAssembly $TESTDRIVE\System.exe
```

Comment:
I believe it is better test module loading in separate sessions to exclude correlations.

Reply:
Module _loading_ **is** done in separate sessions (separate processes). This is just generating the module files to load.